src/sentry/mediators/plugins/migrator.py

```diff
@@ -1,18 +1,19 @@
-from django.db import router
+from typing import List
 
 from sentry.mediators.mediator import Mediator
 from sentry.mediators.param import Param
-from sentry.models import Repository
-from sentry.models.integrations.integration import Integration
 from sentry.plugins.base import plugins
+from sentry.services.hybrid_cloud.integration.model import RpcIntegration
 from sentry.services.hybrid_cloud.organization.model import RpcOrganization
+from sentry.services.hybrid_cloud.repository import repository_service
+from sentry.services.hybrid_cloud.repository.model import RpcRepository
 from sentry.utils.cache import memoize
 
 
 class Migrator(Mediator):
-    integration = Param(Integration)
+    integration = Param(RpcIntegration)
     organization = Param(RpcOrganization)
-    using = router.db_for_write(Integration)
+    using = None
```

Comment:
I think we need this using value to ensure transactions happen on the correct connection.

Reply:
@markstory I couldn't make this work with this transaction.

This mediator just disables plugins via options:

1. https://github.com/getsentry/sentry/blob/9e710e93c0fb6a3e06d25b38514c5e0649b97f9c/src/sentry/mediators/plugins/migrator.py#L36
2. https://github.com/getsentry/sentry/blob/9e710e93c0fb6a3e06d25b38514c5e0649b97f9c/src/sentry/plugins/base/v1.py#L167-L169
3. https://github.com/getsentry/sentry/blob/9e710e93c0fb6a3e06d25b38514c5e0649b97f9c/src/sentry/plugins/base/v1.py#L139-L149
4. https://github.com/getsentry/sentry/blob/9e710e93c0fb6a3e06d25b38514c5e0649b97f9c/src/sentry/plugins/helpers.py#L23-L30

The call to `project_service.update_option()` will violate the invariant that we're calling services within a transaction. 

I've since updated the mediator's parameters to take in `RpcIntegration`. 

In addition, I couldn't find any writes to `Integration`; and hence why I think it's fine to remove the transaction umbrella.

Reply:
Ah, the write being moved to an RPC service would make the transaction incorrect. 👍